### PR TITLE
Run tests by using serverless operator

### PR DESCRIPTION
**Pull request by pierDipi**

### CODE REVIEW

**Comment by pierDipi** [comment]
Notes:
- tracing tests failing in https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_knative-eventing/1752/pull-ci-openshift-knative-eventing-release-v1.3-410-conformance-aws-ocp-410/1531530437896179712
- various `error: error copying source path /go/bin/controller to .: exit status 1` or `error: error copying source path /go/bin/recordevents to .: exit status 1`

File: openshift/release/resolve.sh [hunk 1]
```diff
@@ -23,6 +23,7 @@ function resolve_resources(){
     sed -e "s+\(.* image: \)\(knative.dev\)\(.*/\)\(test/\)\(.*\)+\1\2 \3\4test-\5+g" \
         -e "s+ko://++" \
         -e "s+eventing.knative.dev/release: devel+eventing.knative.dev/release: ${release}+" \
+        -e "s+app.kubernetes.io/version: devel+app.kubernetes.io/version: ${release}+" \
```
Comment (matzew) [inline openshift/release/resolve.sh:26]: +1 nice catch

File: config/core/deployments/webhook-hpa.yaml [hunk 2]
```diff
@@ -49,7 +49,7 @@ metadata:
     app.kubernetes.io/version: devel
     app.kubernetes.io/name: knative-eventing
 spec:
-  minAvailable: 80%
```
[the remark this answers is not in the document]
Comment (pierDipi) [reply config/core/deployments/webhook-hpa.yaml:52]: it's there https://github.com/openshift/knative-eventing/blob/aa45339d9e7e0cb80a9566257f0760a434c707e6/openshift/patches/024-webhook-pdb.patch

**Comment by pierDipi** [comment]
Capturing some notes:

there is this error which needs a fix:
`storage-version-migration-eventing-eventing-1.2.1-sp8l6   0/1   InvalidImageName   0     6m7s`

some other test is failing with:
```
11:23:52.476 INFO:    Create namespaces
Unable to connect to the server: dial tcp: lookup api.ci-ocp-4-9-amd64-aws-us-west-1-px66v.hive.aws.ci.openshift.org on 172.30.0.10:53: no such host
```

File: config/openshift-serverless-view-eventing-configmaps.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  namespace: knative-eventing
+  name: openshift-serverless-view-eventing-configmaps
```
Comment (matzew) [inline config/openshift-serverless-view-eventing-configmaps.yaml:18]: where does this come from?

Comment (matzew) [reply config/openshift-serverless-view-eventing-configmaps.yaml:18]: why do we need to vendor this, than? we we just want to replace the images?

File: openshift/e2e-common.sh [hunk 4]
```diff
@@ -75,146 +75,26 @@ function timeout_non_zero() {
...
-  #oc get pod -n knative-eventing -o yaml | grep image: | grep -v knative-eventing-operator | grep -v ${INTERNAL_REGISTRY} && return 1 || true
-}
-
-function uninstall_knative_eventing(){
-  header "Uninstalling Knative Eventing"
-
-  cat openshift/release/knative-eventing-ci.yaml > ci
-  cat openshift/release/knative-eventing-mtbroker-ci.yaml >> ci
+  local operator_dir=/tmp/serverless-operator
+  git clone --branch replace-eventing-images https://github.com/pierDipi/serverless-operator.git $operator_dir
```
[the remark this answers is not in the document]
Comment (matzew) [reply openshift/e2e-common.sh:85]: So... locally, this PR (with the referenced SO pr) works fine

File: config/openshift-serverless-view-eventing-configmaps.yaml (hunk 3 above)
Comment (pierDipi) [reply config/openshift-serverless-view-eventing-configmaps.yaml:18]: I'm not following, where is the `vendor` involved here?

**Review by matzew — APPROVED** [review]
/lgtm
/approve

**Pull request merged**